shm/Island.hpp

```diff
@@ -0,0 +1,27 @@
+#pragma once
+
+struct Island {
+	struct Coordinates {
+		int positionX;
+		int positionY;
+
+		Coordinates(const int x, const int y):
+			positionX(x),
+			positionY(y)
+		{}
```

Comment:
The definition should be in .cpp files. Only simple getter/setters are allowed in .hpp